internal/stream/stream.go

```diff
@@ -44,7 +44,11 @@ func (ps *PushStream) Start() error {
 		subscriberModAckCh  = make(chan *subscriber.ModAckMessage)
 	)
 
-	ps.subs, err = ps.subscriberCore.NewSubscriber(ps.ctx, ps.nodeID, ps.subscription, 100, 50, 0,
+	// we pass a new context to subscriber because if the subscriber gets a child context of the steam context. there
+	// is a race condition that subscriber exits before the stream is stopped. this causes issues as there are no
+	// subscribers listening to the requests send by stream
+	subscriberCtx := context.Background()
```

Comment:
`subscriber exits before the stream is stopped`: cant we handle this using the `ctx.Timeout()` to avoid a blocking call in case the subscriber channels get closed?

Reply:
ctx.timeout will just wait until timeout, it wont control the exit behaivour of the subscriber

Reply:
Not here. Was suggesting having this at places where we read off like `<-subscriber.channels()`. A timeout will ensure that the request is not blocked forever?

Reply:
that wont work too in this particular case as a select case is already executing, in this particular case our default block was executing, but deriving from the thought process, we shouldn't have a default block. we should move the default processing to a case statement. that way if the case statement is blocked. it will still allow the ctx.Done to execute